src/ui/wallet/ConnectWalletDialog.tsx

```diff
@@ -45,19 +47,29 @@ export function ConnectWalletDialog({
     await activateConnector(walletConnectConnector);
   }, [activateConnector, deactivateActiveConnector]);
 
+  const handleConnectToCoinbase = useCallback(async () => {
```

Comment:
I made this comment before, but we are duplicated logic here. We would have a generic function that takes in a connector as a parameter and handles the deactivation and activation steps.

Reply:
makes sense, i'll make the change !